### PR TITLE
fix(plugins): card component is not KCard [khcp-9970]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

Switch the plugin card component away from using `KCard` since design does not match.
Addresses [KHCP-9970](https://konghq.atlassian.net/browse/KHCP-9970).

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

[KHCP-9970]: https://konghq.atlassian.net/browse/KHCP-9970?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/entities/entities-plugins/sandbox/pages/PluginSelectPage.vue [hunk 1]
```diff
@@ -3,6 +3,7 @@
     <h2>Konnect API</h2>
     <PluginSelect
       :config="konnectConfig"
+      :disabled-plugins="{ 'acl': 'ACL is not supported for this entity type'}"
```
Comment (adamdehaven) [inline packages/entities/entities-plugins/sandbox/pages/PluginSelectPage.vue:6]: question: seems unrelated?

File: packages/entities/entities-plugins/src/components/PluginIcon.vue [hunk 2]
```diff
@@ -31,11 +30,27 @@ defineProps({
   },
 })
 
+const defaultIcon = ref('')
+const src = ref('')
```
Comment (adamdehaven) [inline packages/entities/entities-plugins/src/components/PluginIcon.vue:34]: suggestion: can we rename this different than the native attribute? `imgSrc` or something?

File: packages/entities/entities-plugins/src/components/PluginIcon.vue [hunk 3]
```diff
@@ -31,11 +30,27 @@ defineProps({
...
 const img = ref<HTMLImageElement>()
```
Comment (adamdehaven) [inline packages/entities/entities-plugins/src/components/PluginIcon.vue:35]: img ref should be initialized as `null`

File: packages/entities/entities-plugins/src/components/PluginIcon.vue [hunk 4]
```diff
@@ -31,11 +30,27 @@ defineProps({
...
+    defaultIcon.value = new URL('../assets/images/plugin-icons/missing.png', import.meta.url).href
+  } catch (err) {
+    defaultIcon.value = ''
+  }
+
+  try {
+    src.value = composables.getPluginIconURL(props.name)
+  } catch (err) {
+    src.value = defaultIcon.value
+  }
```
Comment (adamdehaven) [inline packages/entities/entities-plugins/src/components/PluginIcon.vue:54]: question: why isn't the try/catch in the composable where the import is actually happening?

File: packages/entities/entities-plugins/src/components/select/PluginSelectCard.vue [hunk 5]
```diff
@@ -3,66 +3,78 @@
     :label="plugin.disabledMessage"
     position-fixed
   >
-    <KCard
-      class="plugin-card-content"
-      :class="{ disabled: isDisabled }"
+    <div
```
Comment (adamdehaven) [inline packages/entities/entities-plugins/src/components/select/PluginSelectCard.vue:6]: question: why are we using a `div` as a `role="button"`? Can this be an `a` tag instead?

File: packages/entities/entities-plugins/sandbox/pages/PluginSelectPage.vue (hunk 1 above)
Comment (kaiarrowood) [reply packages/entities/entities-plugins/sandbox/pages/PluginSelectPage.vue:6]: Using it to make sure I didn't break display of disabled plugins in the sandbox

File: packages/entities/entities-plugins/src/components/select/PluginSelectCard.vue (hunk 5 above)
Comment (kaiarrowood) [reply packages/entities/entities-plugins/src/components/select/PluginSelectCard.vue:6]: I can switch it, @portikM had recommended `div` with `role="button"` in the associated ticket.

Comment (adamdehaven) [reply packages/entities/entities-plugins/src/components/select/PluginSelectCard.vue:6]: I'd recommend an `a` anchor tag since they are natively focusable and clickable. Leave the `role="button"` unless we're actually navigating (e.g. if we're just opening a modal, or "selecting" something)
